tools/eos-label-location.in

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/python3
+#
+# eos-label-location: edit the location label for metrics instrumentation daemon
+#
+# Copyright (C) 2017 Endless Mobile, Inc.
+# Authors:
+#  Robert McQueen <rob@endlessm.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+import argparse
+import configparser
+import os
+
+class LabelLocation(object):
+    CONFIG_DIR = '@sysconfdir@/metrics/'
+    CONFIG_FILE = CONFIG_DIR + 'location.conf'
+    LABEL_SECTION = 'Label'
+    LABEL_KEYS = ['facility', 'city', 'state']
+
+    def __init__(self, args):
+        self.config = configparser.ConfigParser()
+
+        if args.file:
+            self.CONFIG_DIR = None
+            self.CONFIG_FILE = args.file
+
+        try:
+            self.config.read(self.CONFIG_FILE)
+        except Exception as e:
+            pass
+
+        if self.LABEL_SECTION not in self.config:
+            self.config.add_section(self.LABEL_SECTION)
+
+    def write(self):
+        if self.CONFIG_DIR:
+            os.makedirs(self.CONFIG_DIR, exist_ok=True)
+        with open(self.CONFIG_FILE, 'w') as configfile:
+            self.config.write(configfile)
+
+    def reset(self):
+        for k in self.LABEL_KEYS:
+            self.config.remove_option(self.LABEL_SECTION, k)
+
+        self.write()
+
+    def set(self):
+        for k in self.LABEL_KEYS:
+            old = ''
+            try:
+                old = self.config[self.LABEL_SECTION][k]
+            except KeyError as e:
+                pass
+
+            prompt = 'Enter {}'.format(k)
+            if old != '':
+                prompt += ' [{}]'.format(old)
+            prompt += ': '
+
+            new = input(prompt)
+
+            if new != '':
+                self.config[self.LABEL_SECTION][k] = new
+            elif old == '':
+                self.config.remove_option(self.LABEL_SECTION, k)
+
+        self.write()
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Set an optional label for \
+            the location of this computer, which will be sent to the Endless \
+            OS metrics system.')
+    parser.add_argument('-f', '--file', dest='file',
+                        help='Select config file (default: {})'.format(LabelLocation.CONFIG_FILE))
```

Comment:
it turned out not to be so helpful, because we wish to set a default which it's OK to not exist at runtime and don't wish to create it as side-effect of executing the command with a different -f - but I've used default to good effect now at least